### PR TITLE
Caching support

### PR DESCRIPTION
**Pull request by mmccarty**
This PR adds caching support and implements glob strings for rasterio data sources.

### CODE REVIEW

**Comment by mmccarty** [comment]
@martindurant I don't suppose this PR will go green until we ship intake caching.

**Comment by martindurant** [comment]
I notice that the README is out of data (in fact sort of empty), would appreciate updating it with a minimum of text, perhaps just a link to RTD and Intake.

File: intake_xarray/raster.py [hunk 1]
```diff
@@ -26,21 +28,37 @@ class RasterIOSource(DataSourceMixin):
...
+    def _open_files(self, files):
+        return xr.concat([xr.open_rasterio(f,chunks=self.chunks, **self._kwargs)
+                    for f in files], dim=self.dim)
+
+
     def _open_dataset(self):
-        self._ds = xr.open_rasterio(self.urlpath, chunks=self.chunks,
-                                    **self._kwargs)
+        if '*' in self.urlpath:
+            files = sorted(glob.glob(self.urlpath))
```
[the remark this answers is not in the document]
Comment (mmccarty) [reply intake_xarray/raster.py]: Right.  And we just need to handle lists of urlpaths for now.

File: tests/data/catalog.yaml [hunk 2]
```diff
@@ -24,3 +24,12 @@ sources:
       urlpath: '{{ CATALOG_DIR }}/RGB.byte.tif'
       chunks:
         band: 1
+      concat_dim: concat_dim
+  tiff_glob_source:
```
[the remark this answers is not in the document]
Comment (mmccarty) [reply tests/data/catalog.yaml]: 👍

File: environment.yml [hunk 3]
```diff
@@ -1,104 +1,162 @@
 name: intake-xarray
```
Comment (martindurant) [inline environment.yml]: This is hard to see - only the indentation changed, or something more?

Comment (mmccarty) [reply environment.yml]: Yes, this had gone stale and I was just updating the dependencies.

File: README.md [hunk 4]
```diff
@@ -2,4 +2,21 @@
...
+  - Zarr
+
+### Installation
+
+The conda install instructions are:
+
+```
+pip intall rasterio
+conda install -c conda-forge zarr
+conda install -c intake intake_xarray
```
Comment (martindurant) [inline README.md]: Actually, only the last one is *required*, the other two are required only if you mean to use the specific plugins which reference them.

**Comment by martindurant** [comment]
Looks good from me - were you adding anything else, or is this ready?

**Pull request merged**